src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -397,6 +400,23 @@ static PrivateKey privateKey(Session session, long keyID, String algorithm,
                     }
                 } catch (PKCS11Exception | InvalidKeyException e) {
                     // Attempt failed, create a P11PrivateKey object.
+                    if (debug != null) {
+                        debug.println("Attempt failed, creating a P11PrivateKey object for RSA");
+                    }
+                }
+            }
+        }
+        if((SunPKCS11.mysunpkcs11 != null) && "EC".equals(algorithm)) {
+            if (attributes[0].getBoolean() || attributes[1].getBoolean() || (attributes[2].getBoolean() == false)) {
```

Comment:
Suggest you add
```java
        boolean keySensitive = (attributes[0].getBoolean() ||
        		attributes[1].getBoolean() || !attributes[2].getBoolean());
```
to be used on line 392 and here.

Reply:
Updated